photon-ml/src/main/scala/com/linkedin/photon/ml/algorithm/FactoredRandomEffectCoordinate.scala

```diff
@@ -53,23 +76,35 @@ protected[ml] class FactoredRandomEffectCoordinate[GLM <: GeneralizedLinearModel
   }
 
   /**
-   * Update the model (i.e. run the coordinate optimizer)
+   * Update the coordinate with a new dataset
    *
-   * @param model Rhe model
-   * @return Tuple of updated model and optimization tracker
+   * @param dataSet The updated dataset
+   * @return A new coordinate with the updated dataset
    */
-  protected[algorithm] override def updateModel(model: DatumScoringModel): (DatumScoringModel, OptimizationTracker) =
-    model match {
-      case factoredRandomEffectModel: FactoredRandomEffectModel =>
-        var updatedModelsRDD = factoredRandomEffectModel.modelsInProjectedSpaceRDD
-        var updatedProjectionMatrixBroadcast = factoredRandomEffectModel.projectionMatrixBroadcast
-        var latentFactorOptimizationProblem = factoredRandomEffectOptimizationProblem.latentFactorOptimizationProblem
+  override protected[algorithm] def updateCoordinateWithDataSet(
+    dataSet: RandomEffectDataSet): FactoredRandomEffectCoordinate[RandomFunc, LatentFunc] =
+    new FactoredRandomEffectCoordinate(dataSet, factoredRandomEffectOptimizationProblem)
 
+  /**
+   * Compute an optimized model (i.e. run the coordinate optimizer) for the current dataset using an existing model as
+   * a starting point
+   *
+   * @param model The model to use as a starting point
+   * @return A tuple of the updated model and the optimization states tracker
+   */
+  override protected[algorithm] def updateModel(
+    model: DatumScoringModel): (DatumScoringModel, Option[OptimizationTracker]) = model match {
+      case factoredRandomEffectModel: FactoredRandomEffectModel =>
+        val sparkContext = randomEffectDataSet.sparkContext
+        val randomEffectOptimizationProblem = factoredRandomEffectOptimizationProblem.randomEffectOptimizationProblem
```

Comment:
I've modified this one a little, but in my opinion there's not too much room for change. If you've got a suggestion, I'm open to it